### PR TITLE
Add native analyze e2e tests

### PR DESCRIPTION
**Pull request by karteekmurthys**
Added tests to verify `ANALYZE` command and compare stats between Prestissimo and Presto for the test tables.
Resolves this https://github.com/prestodb/presto/issues/19922

### CODE REVIEW

**Comment by karteekmurthys** [comment]
The `createJavaQueryRunner()` is failing with following exceptions:
```
[ERROR] Failures: 
[ERROR]   TestPrestoNativeAggregations>AbstractTestQueryFramework.init:84->createExpectedQueryRunner:31 ? Creation Unable to create injector, see the following errors:

1) Configuration property 'hive.allow-drop-table' was not used
  at com.facebook.airlift.bootstrap.Bootstrap.lambda$initialize$2(Bootstrap.java:244)

1 error
[ERROR]   TestPrestoNativeJoinQueries>AbstractTestQueryFramework.init:84->createExpectedQueryRunner:31 ? Creation Unable to create injector, see the following errors:

1) Configuration property 'hive.allow-drop-table' was not used
  at com.facebook.airlift.bootstrap.Bootstrap.lambda$initialize$2(Bootstrap.java:244)

1 error
[ERROR]   TestPrestoNativeWindowQueries>AbstractTestQueryFramework.init:84->createExpectedQueryRunner:31 ? Creation Unable to create injector, see the following errors:

1) Configuration property 'hive.allow-drop-table' was not used
  at com.facebook.airlift.bootstrap.Bootstrap.lambda$initialize$2(Bootstrap.java:244)

```

Will try to add hive.allow-drop-table property to the config and test it out.

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java [hunk 1]
```diff
@@ -190,6 +190,29 @@ public void testFiltersAndProjections()
...
+        assertQuery("ANALYZE region");
+        assertQuery("SHOW STATS FOR region");
+        // Create a partitioned table and run analyze on it.
+        String tmpTableName = generateRandomTableName();
+        try {
+            getQueryRunner().execute(getSession(), String.format("CREATE TABLE %s (name VARCHAR, regionkey BIGINT," +
+                    "nationkey BIGINT) WITH (partitioned_by = ARRAY['regionkey','nationkey'])", tmpTableName));
+            Session writeSession = buildSessionForTableWrite();
+            getQueryRunner().execute(writeSession,
+                    String.format("INSERT INTO %s SELECT name, regionkey, nationkey FROM nation", tmpTableName));
```
Comment (aditi-pandit) [inline presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java:205]: "nation" is a very tiny table for this kind of test. But its probably sufficient to test the correctness of this work.

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java [hunk 2]
```diff
@@ -190,6 +190,29 @@ public void testFiltersAndProjections()
...
+            assertQuery(String.format("SELECT * FROM %s", tmpTableName),
+                    "SELECT name, regionkey, nationkey FROM nation");
+            assertQuery("ANALYZE %s WITH (partitions = ARRAY[ARRAY['0','0'],ARRAY['4', '11']])", tmpTableName);
```
Comment (aditi-pandit) [inline presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java:208]: Might be worth adding a comment about the results returned by ANALYZE for this query. Is one of the arrays an empty result ?

**Comment by aditi-pandit** [comment]
Please rebase with https://github.com/prestodb/presto/pull/20025 and check.

**Comment by karteekmurthys** [comment]
> Please rebase with #20025 and check.

Yes, I rebased on it. The tests passed after that. This time hit new failure: https://github.com/prestodb/presto/issues/20029

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java (hunk 2 above)
Comment (karteekmurthys) [reply presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java:216]: I dont' think so. Here are the partitions would look like:

```
presto:tpch> select * from "tmpregion$partitions";
 regionkey | nationkey 
-----------+-----------
         2 |        12 
         2 |         8 
         2 |         9 
         2 |        18 
         2 |        21 
         4 |        13 
         4 |        11 
         4 |        20 
         4 |        10 
         4 |         4 
         3 |        23 
         3 |        22 
         3 |         6 
         3 |         7 
         3 |        19 
         1 |        24 
         1 |         1 
         1 |        17 
         1 |         2 
         1 |         3 
         0 |        15 
         0 |        14 
         0 |         0 
         0 |        16 
         0 |         5 
(25 rows)
```

**Comment by karteekmurthys** [comment]
There seems to be an aggregate function missing required by analyze command:

```
Caused by: java.lang.RuntimeException:  Aggregate function not registered: presto.default.$internal$max_data_size_for_stats
	at com.facebook.presto.tests.AbstractTestingPrestoClient.execute(AbstractTestingPrestoClient.java:124)
	at com.facebook.presto.tests.DistributedQueryRunner.execute(DistributedQueryRunner.java:733)
	at com.facebook.presto.tests.DistributedQueryRunner.execute(DistributedQueryRunner.java:701)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:175)
	... 19 more
```

When I test basic table with a single integer column, analyze works fine on native cluster but if the table has a varchar column, analyze command fails. Looks like I need to create a table with multiple column types like Dates, Varchars, Ints, floats and decimals.

```
presto:kart> insert into t1 values(1),(2),(3),(4);
INSERT: 4 rows
presto:kart> analyze t1;
ANALYZE: 4 rows



presto:kart> insert into t2 values('a'),('b');
INSERT: 2 rows

Query 20230630_042753_00018_2h47r, FINISHED, 2 nodes
Splits: 2 total, 2 done (100.00%)
[Latency: client-side: 0:01, server-side: 0:01] [0 rows, 0B] [0 rows/s, 0B/s]

presto:kart> analyze t2;
Query 20230630_042758_00019_2h47r failed:  Aggregate function not registered: presto.default.$internal$max_data_size_for_stats
```

**Comment by karteekmurthys** [comment]
Closing this PR. Adding tests as part of this https://github.com/prestodb/presto/pull/20055

**Pull request closed without merging**